repos/fdbt-site/src/pages/api/managePassengerGroup.ts

```diff
@@ -136,6 +137,26 @@ export const collectErrors = async (userInput: GroupPassengerTypeDb, nocCode: st
             userInput: userInput.name || '',
         });
     }
+
+    // make sure underlying passenger types are not duplicated
+
+    const fullGroupPassengerType = await convertToFullPassengerType(userInput, nocCode);
+
+    const seen = new Set();
+    let typeName = '';
+
+    const hasDuplicates = fullGroupPassengerType.groupPassengerType.companions.some((item) => {
+        typeName = item.passengerType;
+        return seen.size === seen.add(item.passengerType).size;
+    });
```

Comment:
Minor; if you write this as a `find` and you could remove the mutable variable typeName.